### PR TITLE
PLF-7033 Improve the suggester component

### PR DESCRIPTION
**Pull request by trongtt**
- Handle better the invalid options from Suggester component
- Support customizing option icons easily

### CODE REVIEW

File: commons-extension-webapp/src/main/webapp/javascript/eXo/suggester/suggester.js [hunk 1]
```diff
@@ -400,12 +403,42 @@
...
+                  thumb = settings.defaultOptionIcon.call(this, item, escape);
+                } else {
+                  thumb = settings.defaultOptionIcon;
+                }
+              }
+            }
+
+            return '<div class="option">' +
+              (thumb ? '<img class="thumb" src="' + thumb + '"> ' : '') +
+              escape(item[settings.labelField]) + '</div>';
```
Comment (boubaker) [inline commons-extension-webapp/src/main/webapp/javascript/eXo/suggester/suggester.js]: Are you sure, we should embed the returned DOM here ?
We shouldn't embed the returned DOM sine it's used everywhere, and each usage of Suggester component uses its own classes and DOM.

Comment (trongtt) [inline commons-extension-webapp/src/main/webapp/javascript/eXo/suggester/suggester.js]: The purpose of this is to override the default returned options from origin Selectize component. It helps us customize the option icons easily. You can see the improvement in its usage from https://github.com/exodev/social/pull/179

Moreover, I did care about its compatibility for other usages. So could you please double check @boubaker  ?

Comment (thomasdelhomenie) [inline commons-extension-webapp/src/main/webapp/javascript/eXo/suggester/suggester.js]: Makes sense to have a default common rendering for the platform.

Comment (trongtt) [inline commons-extension-webapp/src/main/webapp/javascript/eXo/suggester/suggester.js]: Yes, default common rendering would help to show option icons easily

**Pull request merged**